src/main/java/com/couchbase/lite/router/Router.java

```diff
@@ -1436,13 +1436,25 @@ public void changed(Database.ChangeEvent event) {
 
             RevisionInternal rev = change.getAddedRevision();
 
+            // https://github.com/couchbase/couchbase-lite-android/issues/683
+            // DocumentChange of Database.ChangeEvent might not contain revision body.
+            // In case of include_docs=true, needs to load revision body
+            if (changesIncludesDocs && (rev.getBody() == null || !rev.getBody().hasContent())) {
+                EnumSet<TDContentOptions> options = getContentOptions();
+                try {
+                    rev = db.loadRevisionBody(rev, options);
+                } catch (CouchbaseLiteException e) {
+                    Log.w(Log.TAG_ROUTER, "Error loading revision body", e);
+                }
+            }
+
```

Comment:
Hi @pasin and @snej,
Thank you very much reviewing code during night. I believe this issue caused by releasing body field aggressively to release memory during pull replication to reduce peek memory usage. It seems that fixing this problem in Router.java is not right approach. I will cancel this PR.
About conflicted document, I will create another ticket to track.
Again Thanks a lot!!